source/ui/memory-component-presenter.cpp

```diff
@@ -62,40 +94,77 @@ int MemoryComponentPresenter::rowCount(const QModelIndex &parent) const {
   return _memorySize;
 }
 
+
 QVariant
 MemoryComponentPresenter::data(const QModelIndex &index, int role) const {
   // check boundaries
   assert::that(index.isValid());
 
-  switch (role) {
-    case AddressRole: {
-      // format index as hex value and return it
-      return QString("%1")
-          .arg(index.row(), 4, 16, QLatin1Char('0'))
-          .toUpper()
-          .prepend("0x");
-    }
-    case ValueRole: {
-      MemoryValue memory_cell =
-          _memoryAccess.getMemoryValueAt(index.row()).get();
-      std::string stringvalue = StringConversions::toHexString(memory_cell);
-      return QString::fromStdString(stringvalue);
-    }
-    case InfoRole: {
-      // TODO fetch value from core
-      return QString("");
-    }
-    default: {
-      // unknown column role, return empty value
-      return QVariant();
-    }
+  // get role as a string because there is more information in it
+  QString role_string = MemoryComponentPresenter::roleNames().value(role);
+
+  int number_of_bits = 8;
+  if (role_string.endsWith("16")) number_of_bits = 16;
+  if (role_string.endsWith("32")) number_of_bits = 32;
+
+  if (role_string.startsWith("address")) {
+    // format index as hex value and return it
+    return QString("%1")
+        .arg(index.row() * number_of_bits / 8, 4, 16, QLatin1Char('0'))
+        .toUpper()
+        .prepend("0x");
   }
+
+  if (role == InfoRole) {
+    // TODO fetch value from core
+    return QString("");
+  }
+
+  MemoryValue memory_cell =
+      _memoryAccess
+          .getMemoryValueAt(index.row() * (number_of_bits / 8),
+                            number_of_bits / 8)
+          .get();
+  std::string stringvalue;
+  if (role_string.startsWith("bin"))
```

Comment:
nein, nein, nein, braces.

The reason is that some Kleinpraktikant might come along and think, hm, I'll add another statement under this `if`:
```
if (role_string.startsWith("bin"))
std::clog << "starts with bin" << std::endl;
stringvalue = StringConversions::toBinString(memory_cell);
```

Question to the audience: what's wrong here?